### ISSUE TITLE
Detection of sharp corners in design patterns

### ISSUE THREAD

**Issue by mawc2019** [NanoComp/photonics-opt-testbed#42]
If a design pattern contains a sharp corner that is not on the boundary, the minimum length scale should be zero or the size of one pixel.
However, as we know, our algorithm for estimating minimum length scales often gives results that are much larger than the size of one pixel. This issue can be partially resolved through the [Harris corner detector](https://docs.opencv.org/3.4/dc/d0d/tutorial_py_features_harris.html). In the function `cornerHarris` in OpenCV, let us set the arguments as `blockSize = 3`, `ksize = 3`, and `k=0.04`. The maxima of `cornerHarris` for a series of wedges are as follows.

Resolution = 100:
![image](https://user-images.githubusercontent.com/60903466/218600462-f187f3da-678c-4f68-bd51-076417ae00cd.png)

Resolution = 10:
![image](https://user-images.githubusercontent.com/60903466/218600524-fbf151d1-0763-4b93-b4ee-5c67c558c203.png)

The relation between the maximum and the angle is as follows. If the threshold is `1e-11`, above which indicates a sharp corner, more than half of the cases here would be correctly indicated.
![image](https://user-images.githubusercontent.com/60903466/218614816-5c757698-f7e2-4a37-ab3e-ef371848e9c4.png)

The maxima of `cornerHarris` for a series of rounded squares are as follows.

Resolution = 100:
![image](https://user-images.githubusercontent.com/60903466/218615566-85aa9285-6b49-42b4-9581-c4e603797720.png)

Resolution = 10:
![image](https://user-images.githubusercontent.com/60903466/218615512-c563fe9a-9e2e-446a-a35b-b3912bd26785.png)

The relation between the maximum and the angle is as follows. If the threshold is `1e-11`, above which indicates a sharp corner, most of the cases here would be correctly indicated.
![image](https://user-images.githubusercontent.com/60903466/218615663-fccce0ff-7d1c-417a-81f7-54a470c5a847.png)

In summary, given the arguments `blockSize = 3`, `ksize = 3`, and `k=0.04`, when the function `cornerHarris` is applied to a binary design pattern and its maximum is higher than the threshold `1e-11`, at least one sharp corners probably exists and the corresponding minimum length scale should be zero or the size of one pixel (which is more proper?). We could add this function to our code and maybe let users decide whether to invoke it.
